app/src/main/java/com/github/se/gatherspot/ui/eventUI/EventView.kt

```diff
@@ -174,36 +174,41 @@ fun EventUINonOrganizer(
               EventRating(eventRating)
 
               Spacer(modifier = Modifier.height(16.dp).testTag("bottomSpacer"))
-
-              // Registration Button
-              // Spacer(modifier = Modifier.height(16.dp))
-
             }
       }
 }
 
 @Composable
-fun AttendButton(event: Event, eventUIViewModel: EventUIViewModel) {
+fun AttendButton(event: Event, eventUIViewModel: EventUIViewModel, status: USER_STATUS) {
   val showDialogAttend by eventUIViewModel.displayAlertAttend.observeAsState()
   val attended by eventUIViewModel.attended.observeAsState()
-  val buttonText = if (attended == true) "Attended" else "Attend"
-  /*
-  when (registrationState) {
-      is RegistrationState.Success -> "Registered"
-      is RegistrationState.Error ->
-          if ((registrationState as RegistrationState.Error).message == "Event is full") "Full"
-          else "Registered"
-      else -> "Register"
+  val buttonText: MutableState<String> = remember {
+    mutableStateOf(
+        if (attended == true && status == USER_STATUS.PARTICIPANT) "Attended"
+        else if (status == USER_STATUS.ORGANIZER) "Verify Users Attendance"
+        else if (attended == false && status == USER_STATUS.PARTICIPANT) "Verify Attendance"
+        else "")
   }
 
-   */
-
   Button(
-      onClick = { eventUIViewModel.attendEvent() },
+      onClick = {
+        if (status == USER_STATUS.PARTICIPANT && attended == false)
+            buttonText.value =
+                "Bring your phone closer to the organizer's phone to verify attendance.\n" +
```

Comment:
this is a lot of text for a button. We should keep it simple: share ticket as text + an icon that symbolizes the action of sharing

Reply:
+ add an interrogation point clickable icon that opens a toaster / dialog to display the explanations about ticket sharing

Reply:
fixed, did not g with the clickable icon because of time constraints